README.md

```diff
@@ -1,69 +1,273 @@
 # reply-urls-operator
-// TODO(user): Add simple overview of use/purpose
+A k8s Operator that keeps Ingress hosts in sync with the Redirect URLs associated with an Azure App Registration.
+
+## Table of Contents
+**[Description](#Description)**<br>
+**[How the Operator works](#How-the-Operator-works)**<br>
+**[Deploying the Operator to a cluster](#Deploying-the-Operator-to-a-cluster)**<br>
+**[Test out the operator locally](#Test-out-the-operator-locally)**<br>
+**[GitHub Workflows](#GitHub-Workflows)**<br>
+**[Modifying the API definitions](#Modifying-the-API-definitions)**<br>
+**[License](#License)**<br>
 
 ## Description
-// TODO(user): An in-depth paragraph about your project and overview of use
+The Reply URLs Operator was created to automate the manual process of updating and removing Redirect URLs when applications are created and removed from AKS clusters. This project aims to follow the Kubernetes [Operator pattern](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/). It uses [Controllers](https://kubernetes.io/docs/concepts/architecture/controller/)
+which provides a reconcile function responsible for synchronizing resources until the desired state is reached on the cluster.
 
 ## Getting Started
-You’ll need a Kubernetes cluster to run against. You can use [KIND](https://sigs.k8s.io/kind) to get a local cluster for testing, or run against a remote cluster.
-**Note:** Your controller will automatically use the current context in your kubeconfig file (i.e. whatever cluster `kubectl cluster-info` shows).
 
-### Running on the cluster
-1. Install Instances of Custom Resources:
+### How the Operator works
+1. Once running, the operator will watch for any Create, Update or Delete events associated with Ingress resources on the cluster it's running on. If you're running the controller locally it will be whichever cluster your kubectl config is pointing to.
+2. When an event occurs on an Ingress on the cluster the operator will act upon that event, depending on the type of event.
+   * **Create/Update:** The Ingress the event is created, will be filtered according the to configuration set in the ReplyURLSync config and will be synced, if it matches the filter and doesn't exist in the list of Reply URLs it will be added.
```

Comment:
```suggestion
   * **Create/Update:** An ingress has been created or updated - will be filtered according to the configuration set in the `ReplyURLSync` config and will be synced, if it matches the filter and doesn't exist in the list of Reply URLs it will be added.
```